### PR TITLE
PMM-7240 DBaaS migration from dashboard to grafana page

### PR DESCRIPTION
**Pull request by tiagomotasantos**
**What this PR does / why we need it**: Migrates DBaaS dashboard to grafana

**Which issue(s) this PR fixes**: https://jira.percona.com/browse/PMM-7240

**Details**:

Main challenges faced doing this migration were types that were missing that had to be created (some were simple but others were more complex and required more time to check where the values were coming from) and components that had to be replaced with platform-core ones (e.g. Modal, RadioButton).

Not a lot of shared components/helper functions were required to move but some were:

- Table
- SelectFieldAdapter
- Overlay
- API helper (it was fixed to work properly with types)
- validators

The use of the API helper will allow us to remove `window.location.origin` when doing HTTP requests and instead do it through this helper.
Also moved the Settings.service so we can use this service from now on instead of defining settings URL in multiple places of the code.

(breadcrumb was not added because PR for the breadcrumb component was not merged yet)
Screenshot 2021-02-05 at 22 48 07

### CODE REVIEW

**Review by fabio-silva — APPROVED** [review]
nice one 🤘

File: public/app/percona/dbaas/components/DBCluster/__mocks__/DBCluster.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { DBCluster, GetDBClustersAction } from '../DBCluster.types';
+
+export const useDBClusters = (kubernetes: Kubernetes[]): [DBCluster[], GetDBClustersAction, boolean] => {
+  const dbClusters: DBCluster[] = [];
+
+  if (kubernetes.length > 0) {
+    dbClusters.push(...dbClustersStub);
+  }
+
+  return [dbClusters, getDBClustersActionStub, false];
```
Comment (atymchuk) [inline public/app/percona/dbaas/components/DBCluster/__mocks__/DBCluster.hooks.ts:12]:
```suggestion
  return [[... dbClustersStub], getDBClustersActionStub, false];
```
And we could get rid of a bunch of code above )

File: public/app/percona/dbaas/components/Kubernetes/Kubernetes.hooks.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  const getKubernetes = async () => {
+    setLoading(true);
+
+    try {
+      const results = (await KubernetesService.getKubernetes()) as KubernetesListAPI;
+
+      setKubernetes(toModelList(results));
+    } catch (e) {
+      console.error(e);
```
Comment (atymchuk) [inline public/app/percona/dbaas/components/Kubernetes/Kubernetes.hooks.ts:31]: Consider the logger?

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/DBClusterBasicOptions/DBClusterBasicOptions.utils.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import React, { FC } from 'react';
+import { SelectableValue } from '@grafana/data';
+import { Messages } from 'app/percona/dbaas/DBaaS.messages';
+import { DatabaseOperators, OPERATORS } from './DBClusterBasicOptions.constants';
+import { KubernetesOperatorStatus } from '../../../Kubernetes/OperatorStatusItem/KubernetesOperatorStatus/KubernetesOperatorStatus.types';
+import { OptionContent } from '../../OptionContent/OptionContent';
+import { Kubernetes } from '../../../Kubernetes/Kubernetes.types';
+import { KubernetesOptionProps } from './DBClusterBasicOptions.types';
+
+export const kubernetesClusterName = (value: string) => {
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/DBClusterBasicOptions/DBClusterBasicOptions.utils.tsx:10]: Done!

File: public/app/percona/dbaas/components/DBCluster/DBClusterService.factory.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { XtraDBService } from './XtraDB.service';
+import { PSMDBService } from './PSMDB.service';
+import { DBClusterServiceDatabasesMap } from './DBCluster.types';
+
+const SERVICE_MAP: Partial<DBClusterServiceDatabasesMap> = {
+  [Databases.mysql]: new XtraDBService(),
+  [Databases.mongodb]: new PSMDBService(),
+};
+
+export class DBClusterServiceFactory {
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/DBCluster/DBClusterService.factory.ts:12]: Done!

File: public/app/percona/dbaas/components/DBCluster/EditDBClusterModal/DBClusterAdvancedOptions/DBClusterAdvancedOptions.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+import { EditDBClusterFields } from '../EditDBClusterModal.types';
+import { DBClusterTopology, DBClusterResources } from './DBClusterAdvancedOptions.types';
+import { resourceValidator } from './DBClusterAdvancedOptions.utils';
+
+export const DBClusterAdvancedOptions: FC<FormRenderProps> = ({ values, form, valid, pristine, submitting }) => {
+  const styles = useStyles(getStyles);
+  const [prevResources, setPrevResources] = useState(DBClusterResources.small);
+  const [customMemory, setCustomMemory] = useState(DEFAULT_SIZES.small.memory);
+  const [customCPU, setCustomCPU] = useState(DEFAULT_SIZES.small.cpu);
+  const { required, min } = validators;
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/DBCluster/EditDBClusterModal/DBClusterAdvancedOptions/DBClusterAdvancedOptions.tsx:25]: No we don't have `min` in core.

File: public/app/percona/dbaas/components/DBCluster/EditDBClusterModal/DBClusterAdvancedOptions/DBClusterAdvancedOptions.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  const { change } = form;
+  const nodeValidators = [required, min(MIN_NODES)];
+  const resourceValidators = [required, min(MIN_RESOURCES), resourceValidator];
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/DBCluster/EditDBClusterModal/DBClusterAdvancedOptions/DBClusterAdvancedOptions.tsx:28]: `MIN_NODES` and `MIN_RESOURCES` are different.

File: public/app/percona/dbaas/components/DBCluster/PSMDB.service.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import { omit } from 'lodash';
+import { Databases } from 'app/percona/shared/core';
+import { apiRequestManagement } from 'app/percona/shared/helpers/api';
```
[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/DBCluster/PSMDB.service.ts:3]: The baseURL here is `/v1/management` hence the `Management` part. Anyway simplified this to just `apiManagement` and removed the `Request` part.

File: public/app/percona/dbaas/components/DBCluster/__mocks__/DBCluster.hooks.ts (hunk 1 above)
Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/DBCluster/__mocks__/DBCluster.hooks.ts:12]: We want to return an empty array when there are no kubernetes, if we remove the rest of the code we won't cover that part.

File: public/app/percona/dbaas/components/Kubernetes/Kubernetes.hooks.ts (hunk 2 above)
Comment (tiagomotasantos) [reply public/app/percona/dbaas/components/Kubernetes/Kubernetes.hooks.ts:31]: Done for all places.

**Review by lunaticusgreen — DISMISSED** [review]
👍

**Review by atymchuk — APPROVED** [review]

**Review by lunaticusgreen — APPROVED** [review]
Great job!

**Pull request merged**